### PR TITLE
[DMU] Yavimaya Iconoclast

### PR DESCRIPTION
**Pull request by FreaIsDead**

### CODE REVIEW

File: Mage.Sets/src/mage/cards/y/YavimayaIconoclast.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+        // Kicker {R}
+        this.addAbility(new KickerAbility("{R}"));
+
+        // Trample
+        this.addAbility(TrampleAbility.getInstance());
+
+        // When Yavimaya Iconoclast enters the battlefield, if it was kicked, it gets +1/+1 and gains haste until end of turn.
+        TriggeredAbility ability = new EntersBattlefieldTriggeredAbility(new BoostSourceEffect(1,1, Duration.EndOfTurn));
+        ability.addEffect(new GainAbilityTargetEffect(HasteAbility.getInstance(), Duration.EndOfTurn));
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/y/YavimayaIconoclast.java:43]: this needs to be `GainAbilitySourceEffect`

File: Mage.Sets/src/mage/cards/y/YavimayaIconoclast.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        ability.addTarget(new TargetPlayer());
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/y/YavimayaIconoclast.java:44]: there's no need for a target added here

File: Mage.Sets/src/mage/cards/y/YavimayaIconoclast.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        this.addAbility(new ConditionalInterveningIfTriggeredAbility(ability, new KickedCostCondition("{R}"),
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/y/YavimayaIconoclast.java:45]: just use `KickedCondition.ONCE`, you don't need the condition that cares about which kicker was used

**Comment by FreaIsDead** [comment]
this should be correct

**Comment by theelk801** [comment]
One last thing, I’m not sure what IDE you’re using but it should have an automatic code formatting feature you can use to clean up imports and reformat code

**Comment by FreaIsDead** [comment]
added on Intellij Setting

**Pull request merged**